Visual Studio Code (1.79.0, undefined, desktop)
Jupyter Extension Version: 2023.4.1001091014.
Python Extension Version: 2023.8.0.
Platform: linux (x64).
Workspace folder ~/study/python/lab, Home = /home/_wajs_err_
03:07:38.898 [info] ZMQ loaded without the fallback mechanism.
03:07:39.211 [debug] Get Custom Env Variables, Class name = wc (started execution), Arg 1: undefined, Arg 2: "RunPythonCode"
03:07:39.214 [info] Start refreshing Kernel Picker (1686355659214)
03:07:39.261 [info] Using Pylance
03:07:39.383 [debug] No controller, hence notebook communications cannot be initialized for editor ~/study/python/lab/lab.ipynb
03:07:40.225 [debug] Get Custom Env Variables, Class name = wc, completed in 1015ms, has a truthy return value, Arg 1: undefined, Arg 2: "RunPythonCode"
03:07:40.225 [debug] Jupyter Paths /kernels: 
03:07:40.242 [debug] Kernel Spec Root Paths, /usr/share/jupyter/kernels, /usr/local/share/jupyter/kernels, ~/.local/share/jupyter/kernels
03:07:40.291 [debug] Getting activated env variables, Class name = Is (started execution), Arg 1: "~/study/python/lab", Arg 2: <Uri:/bin/python>, Arg 3: undefined
03:07:40.292 [debug] Get Custom Env Variables, Class name = wc (started execution), Arg 1: "~/study/python/lab", Arg 2: "RunPythonCode"
03:07:40.296 [debug] Get Custom Env Variables, Class name = wc, completed in 3ms, has a truthy return value, Arg 1: "~/study/python/lab", Arg 2: "RunPythonCode"
03:07:40.314 [debug] Not Searching for kernels as path does not exist, /usr/share/jupyter/kernels
03:07:40.314 [debug] Not Searching for kernels as path does not exist, /usr/local/share/jupyter/kernels
03:07:40.429 [debug] Loading kernelspec from ~/.local/share/jupyter/kernels/python3/kernel.json for 
03:07:40.457 [debug] Conda file returned by Python Extension is conda
03:07:40.483 [warn] Failed to get activated env vars for /bin/python in 190ms
03:07:40.484 [debug] Create ProcessService, Class name = nc (started execution), Arg 1: undefined
03:07:40.484 [debug] Get Custom Env Variables, Class name = wc (started execution), Arg 1: undefined, Arg 2: "RunNonPythonCode", Arg 3: undefined
03:07:40.490 [debug] Get Custom Env Variables, Class name = wc, completed in 6ms, has a truthy return value, Arg 1: undefined, Arg 2: "RunNonPythonCode", Arg 3: undefined
03:07:40.490 [debug] Create ProcessService, Class name = nc, completed in 6ms, has a truthy return value, Arg 1: undefined
03:07:40.501 [info] Process Execution: /bin/python -c "import site;print("USER_BASE_VALUE");print(site.USER_BASE);print("USER_BASE_VALUE");"
03:07:40.587 [debug] USER_SITE for /bin/python is ~/.local/bin
03:07:40.587 [debug] Prepend PATH with user site path for /bin/python, user site ~/.local/bin
03:07:40.587 [debug] Prepend PATH with python bin for /bin/python, 
    PATH value is ~/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl and 
    Path value is undefined
03:07:40.587 [debug] Activated Env Variables for /bin/python, 
    PATH value is /bin:~/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl and 
    Path value is undefined
03:07:40.587 [debug] Got env vars with python /bin/python in 296ms with 46 variables
03:07:40.588 [debug] Getting activated env variables, Class name = Is, completed in 296ms, has a truthy return value, Arg 1: "~/study/python/lab", Arg 2: <Uri:/bin/python>, Arg 3: undefined
03:07:40.598 [info] Process Execution: /bin/python -m pip list
03:07:40.931 [debug] Python API env change detected, update => '/bin/python'
03:07:40.932 [debug] Python API env change detected, update => '/sbin/python'
03:07:40.933 [debug] Python API env change detected, update => '/usr/bin/python'
03:07:40.934 [debug] Python API env change detected, update => '/usr/sbin/python'
03:07:41.118 [debug] Full interpreter list after refreshing (total 167ms, resolve 2ms, refresh 165ms) is length: 4, /bin/python:undefined::/bin/python:/bin/python, /sbin/python:undefined::/sbin/python:/sbin/python, /usr/bin/python:undefined::/usr/bin/python:/usr/bin/python, /usr/sbin/python:undefined::/usr/sbin/python:/usr/sbin/python
03:07:41.124 [debug] refreshData after detecting changes to interpreters
03:07:41.126 [debug] Listing Python & non-Python kernels for Interpreter /bin/python
03:07:41.127 [debug] Listing Python & non-Python kernels for Interpreter /sbin/python
03:07:41.128 [debug] Finding Global Python KernelSpecs
03:07:41.131 [debug] refreshData after detecting changes to interpreters
03:07:41.131 [debug] Listing Python & non-Python kernels for Interpreter /usr/bin/python
03:07:41.134 [debug] refreshData after detecting changes to interpreters
03:07:41.134 [debug] Listing Python & non-Python kernels for Interpreter /usr/sbin/python
03:07:41.135 [debug] refreshData after detecting changes to interpreters
03:07:41.137 [debug] Full interpreter list is length: 4, /bin/python::Unknown:/bin/python, /sbin/python::Unknown:/sbin/python, /usr/bin/python::Unknown:/usr/bin/python, /usr/sbin/python::Unknown:/usr/sbin/python
03:07:41.137 [debug] Refreshed Environments
03:07:41.141 [debug] Kernel for interpreter /sbin/python is .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher
03:07:41.141 [debug] Kernels for interpreter /sbin/python are .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher
03:07:41.149 [debug] Kernel for interpreter /bin/python is .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher
03:07:41.150 [debug] Kernels for interpreter /bin/python are .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher
03:07:41.154 [debug] Kernel for interpreter /usr/bin/python is .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher
03:07:41.154 [debug] Kernels for interpreter /usr/bin/python are .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher
03:07:41.157 [debug] Kernel for interpreter /usr/sbin/python is .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher
03:07:41.158 [debug] Kernels for interpreter /usr/sbin/python are .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher
03:07:41.451 [debug] KernelProvider switched kernel to id = .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher
03:07:41.452 [debug] start the kernel, options.disableUI=true for ~/study/python/lab/lab.ipynb
03:07:41.455 [debug] Initialize notebook communications for editor ~/study/python/lab/lab.ipynb
03:07:41.455 [debug] Resolving notebook UI Comms (resolve) for ~/study/python/lab/lab.ipynb
03:07:41.455 [debug] initialize CommonMessageCoordinator
03:07:41.485 [debug] Registering commtarget jupyter.widget
03:07:41.485 [debug] IPyWidgetMessageDispatcher.initialize
03:07:41.487 [debug] Attempting to determine version of IPyWidgets
03:07:41.490 [debug] Controller selection change completed
03:07:41.490 [debug] Waiting for IPyWidgets version
03:07:41.490 [debug] Waiting for IPyWidgets version promise
03:07:41.492 [info] Starting Kernel startUsingPythonInterpreter, .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher  (Python Path: /bin/python, Unknown, 3.11.3) for '~/study/python/lab/lab.ipynb' (disableUI=true)
03:07:41.493 [debug] Creating raw notebook for resource '~/study/python/lab/lab.ipynb'
03:07:41.495 [debug] Kernel Launcher. launch, Class name = Kg (started execution), Arg 1: ".jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher", Arg 2: 60000, Arg 3: "~/study/python/lab/lab.ipynb", Arg 4: "~/study/python/lab", Arg 5: ""
03:07:41.496 [debug] Getting activated env variables, Class name = Is (started execution), Arg 1: "~/study/python/lab/lab.ipynb", Arg 2: <Uri:/bin/python>, Arg 3: undefined
03:07:41.496 [debug] Get Custom Env Variables, Class name = wc (started execution), Arg 1: "~/study/python/lab/lab.ipynb", Arg 2: "RunPythonCode"
03:07:41.513 [debug] Get Custom Env Variables, Class name = wc, completed in 17ms, has a truthy return value, Arg 1: "~/study/python/lab/lab.ipynb", Arg 2: "RunPythonCode"
03:07:41.526 [debug] Launching kernel in kernelProcess.ts, Class name = Ug (started execution), Arg 1: "~/study/python/lab", Arg 2: ""
03:07:41.945 [warn] Failed to get activated env vars for /bin/python in 449ms
03:07:41.946 [debug] Prepend PATH with user site path for /bin/python, user site ~/.local/bin
03:07:41.946 [debug] Prepend PATH with python bin for /bin/python, 
    PATH value is ~/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl and 
    Path value is undefined
03:07:41.946 [debug] Activated Env Variables for /bin/python, 
    PATH value is /bin:~/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl and 
    Path value is undefined
03:07:41.946 [debug] Got env vars with python /bin/python in 450ms with 46 variables
03:07:41.946 [debug] Getting activated env variables, Class name = Is, completed in 450ms, has a truthy return value, Arg 1: "~/study/python/lab/lab.ipynb", Arg 2: <Uri:/bin/python>, Arg 3: undefined
03:07:41.962 [info] Process Execution: /bin/python -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
03:07:41.962 [debug] Kernel Env Variables for ~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/temp/jupyter/kernels/pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a/kernel.json, PATH value is /bin:~/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
03:07:41.977 [info] Process Execution: /bin/python -m ipykernel_launcher --ip=127.0.0.1 --stdin=9003 --control=9001 --hb=9000 --Session.signature_scheme="hmac-sha256" --Session.key=b"f0fa2356-3e63-470b-9b7e-363af7bcc90e" --shell=9002 --transport="tcp" --iopub=9004 --f=~/.local/share/jupyter/runtime/kernel-v2-973xCw8jth0K3uq.json
    > cwd: ~/study/python/lab
03:07:41.977 [debug] Launching kernel in kernelProcess.ts, Class name = Ug, completed in 451ms, has a truthy return value, Arg 1: "~/study/python/lab", Arg 2: ""
03:07:42.001 [warn] Stderr output when getting ipykernel version & path Traceback (most recent call last):
    File "<string>", line 1, in <module>
    ModuleNotFoundError: No module named 'ipykernel' for /bin/python
03:07:42.007 [debug] KernelProcess error: /bin/python: No module named ipykernel_launcher

03:07:42.007 [warn] StdErr from Kernel Process /bin/python: No module named ipykernel_launcher
03:07:42.011 [debug] KernelProcess Exit Exit - 1 /bin/python: No module named ipykernel_launcher

03:07:42.012 [error] Disposing kernel process due to an error [mp [Error]: The kernel died. Error: /bin/python: No module named ipykernel_launcher... View Jupyter [log](command:jupyter.viewOutput) for further details.
	at ChildProcess.<anonymous> (~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/out/extension.node.js:24:86433)
	at ChildProcess.emit (node:events:525:35)
	at ChildProcess._handle.onexit (node:internal/child_process:291:12)] {
  category: 'kerneldied',
  isJupyterError: true,
  kernelConnectionMetadata: Ku {
    kind: 'startUsingPythonInterpreter',
    kernelSpec: lu {
      specFile: '~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/temp/jupyter/kernels/pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a/kernel.json',
      interpreterPath: '/bin/python',
      isRegisteredByVSC: 'registeredByNewVersionOfExt',
      name: 'pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a',
      argv: [Array],
      language: 'python',
      executable: 'python',
      display_name: 'Python 3',
      metadata: [Object],
      env: {},
      interrupt_mode: undefined
    },
    interpreter: {
      id: '/bin/python',
      sysPrefix: '/usr',
      envPath: undefined,
      displayPath: [n],
      envName: '',
      uri: [n],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher'
  },
  exitCode: 1,
  stdErr: '/bin/python: No module named ipykernel_launcher\n'
}
03:07:42.014 [info] Dispose Kernel process 1183.
03:07:42.017 [debug] Disposed Kernel process 1183.
03:07:42.021 [error] Kernel Launcher. launch, Class name = Kg, completed in 526ms, has a falsy return value, Arg 1: ".jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher", Arg 2: 60000, Arg 3: "~/study/python/lab/lab.ipynb", Arg 4: "~/study/python/lab", Arg 5: "" [mp [Error]: The kernel died. Error: /bin/python: No module named ipykernel_launcher... View Jupyter [log](command:jupyter.viewOutput) for further details.
	at ChildProcess.<anonymous> (~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/out/extension.node.js:24:86433)
	at ChildProcess.emit (node:events:525:35)
	at ChildProcess._handle.onexit (node:internal/child_process:291:12)] {
  category: 'kerneldied',
  isJupyterError: true,
  kernelConnectionMetadata: Ku {
    kind: 'startUsingPythonInterpreter',
    kernelSpec: lu {
      specFile: '~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/temp/jupyter/kernels/pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a/kernel.json',
      interpreterPath: '/bin/python',
      isRegisteredByVSC: 'registeredByNewVersionOfExt',
      name: 'pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a',
      argv: [Array],
      language: 'python',
      executable: 'python',
      display_name: 'Python 3',
      metadata: [Object],
      env: {},
      interrupt_mode: undefined
    },
    interpreter: {
      id: '/bin/python',
      sysPrefix: '/usr',
      envPath: undefined,
      displayPath: [n],
      envName: '',
      uri: [n],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher'
  },
  exitCode: 1,
  stdErr: '/bin/python: No module named ipykernel_launcher\n'
}
03:07:42.022 [error] Failed to connect raw kernel session: Error: The kernel died. Error: /bin/python: No module named ipykernel_launcher... View Jupyter [log](command:jupyter.viewOutput) for further details.
03:07:42.022 [debug] Shutdown session -- complete
03:07:42.022 [debug] failed to create IJupyterKernelConnectionSession in kernel, UI Disabled = true [mp [Error]: The kernel died. Error: /bin/python: No module named ipykernel_launcher... View Jupyter [log](command:jupyter.viewOutput) for further details.
	at ChildProcess.<anonymous> (~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/out/extension.node.js:24:86433)
	at ChildProcess.emit (node:events:525:35)
	at ChildProcess._handle.onexit (node:internal/child_process:291:12)] {
  category: 'kerneldied',
  isJupyterError: true,
  kernelConnectionMetadata: Ku {
    kind: 'startUsingPythonInterpreter',
    kernelSpec: lu {
      specFile: '~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/temp/jupyter/kernels/pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a/kernel.json',
      interpreterPath: '/bin/python',
      isRegisteredByVSC: 'registeredByNewVersionOfExt',
      name: 'pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a',
      argv: [Array],
      language: 'python',
      executable: 'python',
      display_name: 'Python 3',
      metadata: [Object],
      env: {},
      interrupt_mode: undefined
    },
    interpreter: {
      id: '/bin/python',
      sysPrefix: '/usr',
      envPath: undefined,
      displayPath: [n],
      envName: '',
      uri: [n],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher'
  },
  exitCode: 1,
  stdErr: '/bin/python: No module named ipykernel_launcher\n'
}
03:07:42.026 [warn] Error occurred while trying to start the kernel, options.disableUI=true [mp [Error]: The kernel died. Error: /bin/python: No module named ipykernel_launcher... View Jupyter [log](command:jupyter.viewOutput) for further details.
	at ChildProcess.<anonymous> (~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/out/extension.node.js:24:86433)
	at ChildProcess.emit (node:events:525:35)
	at ChildProcess._handle.onexit (node:internal/child_process:291:12)] {
  category: 'kerneldied',
  isJupyterError: true,
  kernelConnectionMetadata: Ku {
    kind: 'startUsingPythonInterpreter',
    kernelSpec: lu {
      specFile: '~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/temp/jupyter/kernels/pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a/kernel.json',
      interpreterPath: '/bin/python',
      isRegisteredByVSC: 'registeredByNewVersionOfExt',
      name: 'pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a',
      argv: [Array],
      language: 'python',
      executable: 'python',
      display_name: 'Python 3',
      metadata: [Object],
      env: {},
      interrupt_mode: undefined
    },
    interpreter: {
      id: '/bin/python',
      sysPrefix: '/usr',
      envPath: undefined,
      displayPath: [n],
      envName: '',
      uri: [n],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher'
  },
  exitCode: 1,
  stdErr: '/bin/python: No module named ipykernel_launcher\n'
}
03:07:42.138 [info] End refreshing Kernel Picker (1686355659214)
03:07:55.380 [info] Shutting down notebooks for 3206f15b-eb84-467c-96f0-e9096e9819a1
03:07:55.380 [info] Disposing HostJupyterExecution 532fc0b1-a179-45b4-9594-7b03b9808a3d
03:07:55.380 [debug] Disposing super HostJupyterExecution 532fc0b1-a179-45b4-9594-7b03b9808a3d
03:07:55.380 [info] Dispose Kernel '~/study/python/lab/lab.ipynb' associated with '~/study/python/lab/lab.ipynb'
03:07:55.381 [debug] Kernel got disposed, hence there is no longer a kernel associated with ~/study/python/lab/lab.ipynb
03:07:55.381 [info] Disposing HostJupyterExecution 532fc0b1-a179-45b4-9594-7b03b9808a3d
03:07:55.381 [debug] Finished disposing HostJupyterExecution  532fc0b1-a179-45b4-9594-7b03b9808a3d
03:07:55.383 [debug] Disposing controller .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher associated with connection .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher 
03:07:55.389 [debug] Disposing controller .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher (Interactive) associated with connection .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher 
03:07:55.390 [debug] Disposing controller .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher associated with connection .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher and documents file://~/study/python/lab/lab.ipynb
03:07:55.390 [debug] Disposing controller .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher (Interactive) associated with connection .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher 
03:07:55.390 [debug] Disposing controller .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher associated with connection .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher 
03:07:55.390 [debug] Disposing controller .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher (Interactive) associated with connection .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher 
03:07:55.390 [debug] Disposing controller .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher associated with connection .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher 
03:07:55.390 [debug] Disposing controller .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher (Interactive) associated with connection .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher 
03:07:55.391 [debug] Cleaning up server cache 532fc0b1-a179-45b4-9594-7b03b9808a3d
03:07:55.392 [debug] Finished disposing HostJupyterExecution  532fc0b1-a179-45b4-9594-7b03b9808a3d
